cli/get.go

```diff
@@ -131,25 +150,51 @@ A role must be specified when using this command through the --role flag. You ma
 			return echoCredentials(args[0], args[0], credentials, outputType, shellType, awsCliPath, tencentCliPath)
 		}
 
-		oauthCfg, err := DiscoverOAuth2Config(cmd.Context(), oidcDomain, clientID)
-		if err != nil {
-			cmd.PrintErrf("could not discover oauth2  config: %s\n", err)
-			return nil
-		}
+		var assertionBytes []byte
+		switch cloudType {
+		case cloudAws:
+			oauthCfg, err := DiscoverOAuth2Config(cmd.Context(), oidcDomain, clientID)
+			if err != nil {
+				cmd.PrintErrf("could not discover oauth2  config: %s\n", err)
+				return nil
+			}
 
-		tok, err := ExchangeAccessTokenForWebSSOToken(cmd.Context(), client, oauthCfg, config.Tokens, account.ID)
-		if err != nil {
-			cmd.PrintErrf("error exchanging token: %s\n", err)
-			return nil
-		}
+			tok, err := ExchangeAccessTokenForWebSSOToken(cmd.Context(), client, oauthCfg, config.Tokens, account.ID)
+			if err != nil {
+				cmd.PrintErrf("error exchanging token: %s\n", err)
+				return nil
+			}
 
-		assertion, err := ExchangeWebSSOTokenForSAMLAssertion(cmd.Context(), client, oidcDomain, tok)
-		if err != nil {
-			cmd.PrintErrf("failed to fetch SAML assertion: %s\n", err)
+			assertionBytes, err = ExchangeWebSSOTokenForSAMLAssertion(cmd.Context(), client, oidcDomain, tok)
+			if err != nil {
+				cmd.PrintErrf("failed to fetch SAML assertion: %s\n", err)
+				return nil
+			}
+		case cloudTencent:
+			// Tencent applications aren't supported by the Okta API, so we can't use the same flow as AWS.
+			// Instead, we can construct a URL to initiate logging into the application that has been pre-configured to support KeyConjurer.
+			// This URL will redirect back to a web server known ahead of time with a SAML assertion which we can then exchange for credentials.
+			//
+			// Because we need to know the URL of the application, --bypass-cache can't be used.
+			if bypassCache {
+				cmd.PrintErrf("cannot use --%s with Tencent applications\n", FlagBypassCache)
+				return nil
+			}
+
+			// TODO: Spin up a web server that listens for a SAML callback.
+
+			// TODO: This only works for OSX.
```

Comment:
We can use the work we have to implement this to add a new flag, `-b/--open-browser` and add that to `keyconjurer login` so users don't need to copy/paste.